test/SILGen/moveonly.swift

```diff
@@ -1000,3 +1004,2127 @@ struct AddressOnlySetterTester : ~Copyable {
+// CHECK: [[COPYABLE_CLASS:%.*]] = load [copy] [[ACCESS]]
+// CHECK: end_access [[ACCESS]]
+// CHECK: [[BORROW_COPYABLE_CLASS:%.*]] = begin_borrow [[COPYABLE_CLASS]]
+// CHECK: ([[CORO_RESULT:%.*]], [[CORO_TOKEN:%.*]]) = begin_apply {{%.*}}([[BORROW_COPYABLE_CLASS]])
+// CHECK: ([[CORO_RESULT_2:%.*]], [[CORO_TOKEN_2:%.*]]) = begin_apply {{%.*}}({{%.*}}, [[CORO_RESULT]])
+// CHECK: [[TEMP:%.*]] = alloc_stack $
+// CHECK: copy_addr [[CORO_RESULT_2]] to [init] [[TEMP]]
+// CHECK: end_apply [[CORO_TOKEN_2]]
+// CHECK: end_apply [[CORO_TOKEN]]
+// CHECK: end_borrow [[BORROW_COPYABLE_CLASS]]
+// CHECK: destroy_value [[COPYABLE_CLASS]]
+//
+// Fourth Mutation
+// CHECK: [[ACCESS:%.*]] = begin_access [read] [unknown] [[PROJECT]]
+// CHECK: [[COPYABLE_CLASS:%.*]] = load [copy] [[ACCESS]]
+// CHECK: end_access [[ACCESS]]
+// CHECK: [[BORROW_COPYABLE_CLASS:%.*]] = begin_borrow [[COPYABLE_CLASS]]
+// CHECK: ([[CORO_RESULT:%.*]], [[CORO_TOKEN:%.*]]) = begin_apply {{%.*}}([[BORROW_COPYABLE_CLASS]])
+// CHECK: [[TEMP:%.*]] = alloc_stack $
+// CHECK: [[MARK_TEMP:%.*]] = mark_must_check [consumable_and_assignable] [[TEMP]]
+// CHECK: [[LOAD:%.*]] = load_borrow [[CORO_RESULT]]
+// CHECK: ([[CORO_RESULT_2:%.*]], [[CORO_TOKEN_2:%.*]]) = begin_apply {{%.*}}({{%.*}}, [[LOAD]])
+// CHECK: copy_addr [[CORO_RESULT_2]] to [init] [[MARK_TEMP]]
+// CHECK: end_apply [[CORO_TOKEN_2]]
+// CHECK: end_borrow [[LOAD]]
+// Mutating Func
+// CHECK: apply {{%.*}}([[MARK_TEMP]])
+// Setter
+// CHECK: apply {{%.*}}([[MARK_TEMP]], {{%.*}}, [[CORO_RESULT]])
+// CHECK: end_apply [[CORO_TOKEN]]
+// CHECK: } // end sil function '$s8moveonly73testSubscriptReadSetThroughParentClass_BaseLoadable_ResultAddressOnly_VaryyF'
+public func testSubscriptReadSetThroughParentClass_BaseLoadable_ResultAddressOnly_Var() {
+    var m = LoadableSubscriptReadSetTesterClassParent()
+    m = LoadableSubscriptReadSetTesterClassParent()
+    m.tester[0].nonMutatingFunc()
+    m.tester[0].mutatingFunc()
+    m.testerParent.tester[0].nonMutatingFunc()
+    m.testerParent.tester[0].mutatingFunc()
+    m.computedTester[0].nonMutatingFunc()
+    m.computedTester2[0].nonMutatingFunc()
+    m.computedTester2[0].mutatingFunc()
+}
+
+// MARK: _read and _modify
+
+public struct LoadableSubscriptReadModifyTester : ~Copyable {
+    subscript(_ i: Int) -> AddressOnlyProtocol {
+        _read {
+            fatalError()
+        }
+        _modify {
+            fatalError()
+        }
+    }
+}
+
+// CHECK-LABEL: sil [ossa] @$s8moveonly58testSubscriptReadModify_BaseLoadable_ResultAddressOnly_VaryyF : $@convention(thin) () -> () {
+// CHECK: [[BOX:%.*]] = alloc_box $
+// CHECK: [[PROJECT:%.*]] = project_box [[BOX]]
+//
+// The read call
+// CHECK: [[ACCESS:%.*]] = begin_access [read] [unknown] [[PROJECT]]
+// CHECK: [[MARK:%.*]] = mark_must_check [no_consume_or_assign] [[ACCESS]]
+// CHECK: [[LOAD:%.*]] = load_borrow [[MARK]]
+// Eventually, we need this end_apply to be around the nonMutatingFunc.
```

Comment:
Since `AddressOnlyProtocol` is move-only, we can't copy it out of the `end_apply`/`end_borrow`/`end_access`, and the call to `nonMutatingMethod` does have to happen during the borrow. So "eventually" doesn't seem to cut it. Does the move only checker fix this up?

Reply:
It was an issue where SILGen was failing to pattern match. I did an update that fixes this.